File: src/AddIns/Misc/PackageManagement/Project/Src/EnvDTE/CodeTypeRef2.cs
// Copyright (c) AlphaSierraPapa for the SharpDevelop Team (for details please see \doc\copyright.txt)
// This code is distributed under the GNU LGPL (for details please see \doc\license.txt)

using System;
using ICSharpCode.NRefactory.TypeSystem;
using ICSharpCode.SharpDevelop.Dom;

namespace ICSharpCode.PackageManagement.EnvDTE
{
	public class CodeTypeRef2 : CodeTypeRef, global::EnvDTE.CodeTypeRef2
	{
		public CodeTypeRef2(CodeModelContext context, CodeElement parent, IType type)
			: base(context, parent, type)
		{
		}
		
		public bool IsGeneric {
			get { return type.IsParameterized; }
		}
	}
}

File: src/AddIns/Misc/PackageManagement/SharpDevelop.EnvDTE/Src/CodeTypeRef2.vb
' Copyright (c) AlphaSierraPapa for the SharpDevelop Team (for details please see \doc\copyright.txt)
' This code is distributed under the GNU LGPL (for details please see \doc\license.txt)

Namespace EnvDTE
	Public Interface CodeTypeRef2
		Inherits CodeTypeRef
		
		ReadOnly Property IsGeneric As Boolean
	End Interface
End Namespace